modules/ROOT/generate-docs.sh

```diff
@@ -2,14 +2,42 @@
 
 set -exuo pipefail
 
-cd $(dirname $0)
-
 GIT_ROOT=$(git rev-parse --show-toplevel)
 
-PATH_TO_SYNC_GATEWAY=${PATH_TO_SYNC_GATEWAY:-${GIT_ROOT}/../sync_gateway/}
+# Get upstream sync_gateway repo
+cd "${GIT_ROOT}"
+branch=release/3.2.1
+
+[[ -d ./sync_gateway ]] || git clone --no-checkout https://github.com/couchbase/sync_gateway.git
+cd sync_gateway
+git sparse-checkout init --cone
+git sparse-checkout set docs/api
+git checkout ${branch}
+git pull --depth=1 origin ${branch}
+cd ..
+
+PATH_TO_SYNC_GATEWAY=${PATH_TO_SYNC_GATEWAY:-${GIT_ROOT}/sync_gateway}
 PATH_TO_SPECS="${PATH_TO_SYNC_GATEWAY}/docs/api/"
 
-PATH_TO_BUNDLE=${PATH_TO_BUNDLE:-${GIT_ROOT}/modules/ROOT/assets/attachments/}
+PATH_TO_BUNDLE=${PATH_TO_BUNDLE:-${GIT_ROOT}/modules/ROOT/assets/attachments}
+PATH_TO_OVERLAY=${PATH_TO_OVERLAY:-${GIT_ROOT}/modules/ROOT/assets/overlays}
+PATH_TO_TEMP=${PATH_TO_TEMP:-${GIT_ROOT}/modules/ROOT/assets/bld}
+
+# Create a bld directory to hold temporary files, if it doesn't already exist
+[[ -d "${PATH_TO_TEMP}" ]] || mkdir "${PATH_TO_TEMP}"
+
+# Use jq to transform the Prometheus metrics file into JSON schema.
```

Comment:
I wonder if you should auto generate this file. The way to generate this would be to `cd sync_gateway/tools/stats-defintion-exporter && go run .` and this will output a current `metrics_metadata.json` in the cwd.

this might be easier than grabbing this from latestbuilds, especially because you already have the repo checked out here.

Reply:
Ah, that's interesting, thanks! I was wondering how that file was generated. Currently we rely on extracting that file from a build package by hand.

For the purposes of this PR, I think we can assume that `metrics_metadata.json` exists and is up-to-date, because it is already used to generate the static Prometheus metrics documentation in [this page](https://docs.couchbase.com/sync-gateway/current/stats-monitoring-prometheus.html). But I'll talk to @osfameron about the process of generating that page and see if we can incorporate your method.

Reply:
Answer from @osfameron was "no" to this suggestion:

> I'd prefer to consume developer artefacts, not generate them
> learning how to build Go dependencies and then periodically maintaining that toolchain when the upstream devs change things adds no value to docs toolchain.